### PR TITLE
Add flip() method and proptests

### PR DESCRIPTION
**Pull request by rhagenson**
Provides a `flip()` method to either add or remove a value and returns whether the value was present prior to the flip.

### CODE REVIEW

File: roaring/test/main.pony [hunk 1]
```diff
@@ -38,6 +40,27 @@ class SetTwice is Property1[U32]
...
+    h.assert_true(roaring.contains(arg1))  // Does contain value
+
+class FlipTwice is Property1[U32]
+  fun name(): String => "flip() is false, then flip is true"
+
+  fun gen(): Generator[U32] => Generators.u32(U32.min_value(), U32.max_value())
+
+  fun property(arg1: U32, h: PropertyHelper) =>
+    let roaring = Roaring
+    h.assert_false(roaring.set(arg1))  // Value not previously set
```
Comment (mfelsche) [inline roaring/test/main.pony]: this should be a call to flip, right?

**Review by mfelsche — COMMENTED** [review]
LGTM, just a little test typo

File: roaring/test/main.pony (hunk 1 above)
Comment (rhagenson) [reply roaring/test/main.pony]: Yes, I will get that fixed. Bit of copy-paste-forget going on there.

**Comment by rhagenson** [comment]
I will get the typo fixed later today when I back at my home workstation.

**Review by mfelsche — APPROVED** [review]

**Comment by mfelsche** [comment]
This needs some conflict resolving with the unset changes.

**Comment by rhagenson** [comment]
Unless you want to make the changes, I can get the conflict fixed this evening.

If I fix the conflict, any issue with me merging since I would not be changing content?

**Comment by mfelsche** [comment]
Go ahead merging! :) :+1:

**Pull request merged**